File: ProgrammersAlgorithm/JS/Level_3/2xNTiling.js
function solution(n) {
  let d = Array.from({ length: 60001 }, () => 0);

  d[1] = 1;
  d[2] = 2;

  for (let i = 3; i <= n; i++) {
    d[i] = (d[i - 1] + d[i - 2]) % 1000000007;
  }

  return d[n];
}

File: ProgrammersAlgorithm/Cpp/Level_3/2xNTiling.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    vector<int> d(n+1,0);

    d[1] = 1;
    d[2] = 2;

    for(int i=3;i<n+1;i++) {
        d[i] = (d[i-2] + d[i-1])%1000000007;
    }

    return d[n];
}